polaris/dataset/_base.py

```diff
@@ -294,13 +323,20 @@ def to_json(
         """
         raise NotImplementedError
 
-    def cache(self) -> str:
+    def cache(self, verify_checksum: bool = True) -> str:
         """Caches the dataset by downloading all additional data for pointer columns to a local directory.
 
+        Args:
+            verify_checksum: Whether to verify the checksum of the dataset after caching.
+
         Returns:
             The path to the cache directory.
         """
         self.to_json(self._cache_dir, load_zarr_from_new_location=True)
+
+        if verify_checksum:
+            self.verify_checksum()
```

Comment:
Comment: Same comment here. Verifying the checksum only makes sense for the V1 and shouldn't be in the base class.